internal/providers/providers.go

```diff
@@ -287,11 +290,44 @@ func getCredentialForProvider(
 	}
 	if errors.Is(err, sql.ErrNoRows) {
 		zerolog.Ctx(ctx).Debug().Msg("no access token found for provider")
-		return credentials.NewEmptyCredential(), nil
+
+		// If we don't have an access token, check if we have an installation ID
+		return getInstallationTokenCredential(ctx, prov, store)
 	}
 	return nil, fmt.Errorf("error getting credential: %w", err)
 }
 
+// getInstallationTokenCredential returns a GitHub installation token credential if the provider has an installation ID
+func getInstallationTokenCredential(ctx context.Context, prov db.Provider, store db.Store) (provinfv1.Credential, error) {
+	installation, err := store.GetInstallationIDByProviderID(ctx, uuid.NullUUID{
+		UUID:  prov.ID,
+		Valid: true,
+	})
+	if err == nil {
```

Comment:
nit: Let's first check no rows, then throw the error and do the logic in the bottom of the function. That way we reduce indentation and cognitive load.

Reply:
Done!